casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -190,60 +190,56 @@ object Validate {
-                      )
-      _ <- span.mark("before-sequence-number-validation")
-      sequenceNumberStatus <- parentsStatus.joinRight.traverse(
-                               _ => Validate.sequenceNumber[F](block, dag)
-                             )
-      _ <- span.mark("before-justification-regression-validation")
-      justificationRegressionsStatus <- sequenceNumberStatus.joinRight.traverse(
-                                         _ =>
-                                           Validate.justificationRegressions[F](block, genesis, dag)
-                                       )
-      _ <- span.mark("before-shrad-identifier-validation")
-      shardIdentifierStatus <- justificationRegressionsStatus.joinRight.traverse(
-                                _ => Validate.shardIdentifier[F](block, shardId)
-                              )
-    } yield shardIdentifierStatus.joinRight
+  ): F[Either[BlockStatus, ValidBlock]] = {
+    val getBlockSummary: EitherT[F, BlockStatus, ValidBlock] = for {
+      _ <- EitherT.liftF(span.mark("before-block-hash-validation"))
+      _ <- EitherT(Validate.blockHash[F](block))
+      _ <- EitherT.liftF(span.mark("before-deploy-count-validation"))
+      _ <- EitherT(Validate.deployCount[F](block))
+      _ <- EitherT.liftF(span.mark("before-missing-blocks-validation"))
+      _ <- EitherT(Validate.missingBlocks[F](block, dag))
+      _ <- EitherT.liftF(span.mark("before-timestamp-validation"))
+      _ <- EitherT(Validate.timestamp[F](block, dag))
+      _ <- EitherT.liftF(span.mark("before-valid-deploy-signatures"))
+      _ <- validDeploySignatures[F](block)
+      _ <- EitherT.liftF(span.mark("before-repeat-deploy-validation"))
+      _ <- EitherT(Validate.repeatDeploy[F](block, dag, expirationThreshold, span))
+      _ <- EitherT.liftF(span.mark("before-block-number-validation"))
+      _ <- EitherT(Validate.blockNumber[F](block, dag))
+      _ <- EitherT.liftF(span.mark("before-future-transaction-validation"))
+      _ <- EitherT(Validate.futureTransaction[F](block))
+      _ <- EitherT.liftF(span.mark("before-transaction-expired-validation"))
+      _ <- EitherT(Validate.transactionExpiration[F](block, expirationThreshold))
+      _ <- EitherT.liftF(span.mark("before-justification-follows-validation"))
+      _ <- EitherT(Validate.justificationFollows[F](block, genesis, dag))
+      _ <- EitherT.liftF(span.mark("before-parents-validation"))
+      _ <- EitherT(Validate.parents[F](block, genesis, dag))
+      _ <- EitherT.liftF(span.mark("before-sequence-number-validation"))
+      _ <- EitherT(Validate.sequenceNumber[F](block, dag))
+      _ <- EitherT.liftF(span.mark("before-justification-regression-validation"))
+      _ <- EitherT(Validate.justificationRegressions[F](block, genesis, dag))
+      _ <- EitherT.liftF(span.mark("before-shrad-identifier-validation"))
+      _ <- EitherT(Validate.shardIdentifier[F](block, shardId))
+    } yield Valid
```

Comment:
isn't this a case where the operator `>>` would make the code cleaner than the for comprehension?

Reply:
it potentially would, matter of style